web/frontend/src/schema/configurationValidation.ts

```diff
@@ -0,0 +1,400 @@
+        }
+
+        if (MaxN < 1) {
+          return this.createError({
+            path,
+            message: `Max should be higher or equal to 1 in selects, in object ID: ${ID}`,
+          });
+        }
+
+        if (MaxN < MinN) {
+          return this.createError({
+            path,
+            message: `Max should be higher or equal than min in selects, in object ID: ${ID}`,
+          });
+        }
+
+        if (MaxN >= Choices.length) {
+          return this.createError({
+            path,
+            message: `MaxN should be less or equal to Choices length in selects, in object ID: ${ID}`,
+          });
+        }
+        return true;
+      },
+    })
+    .required(),
+  MinN: yup
+    .number()
+    .max(yup.ref('MaxN'), `Min should be smaller than max in selects`)
+    .test({
+      name: 'compare-min-max',
+      message: 'Max/Min comparison failed',
+      test() {
+        const { path, parent } = this;
+        const { MinN, ID } = parent;
+
+        if (!Number.isInteger(MinN)) {
+          return this.createError({
+            path,
+            message: `Min should be an integer in selects, in object ID: ${ID}`,
+          });
+        }
+
+        if (MinN < 1) {
+          return this.createError({
+            path,
+            message: `Min should be higher or equal to 1 in selects, in object ID: ${ID}`,
+          });
+        }
+        return true;
+      },
+    })
+    .required(),
+  Choices: yup
+    .array()
+    .of(yup.string())
+    .test({
+      name: 'compare-choices-min-max',
+      message: 'Choice length comparison failed',
+      test() {
+        const { path, parent } = this;
+        const { MaxN, Choices, ID } = parent;
+
+        if (Choices.length < MaxN) {
+          return this.createError({
+            path,
+            message: `Choices array length should be at least equal to Max in selects, in object ID: ${ID}`,
+          });
+        }
+        if (Choices.includes('')) {
+          return this.createError({
+            path,
+            message: `Choices array should not contain empty strings in selects, in object ID: ${ID}`,
+          });
+        }
+        return true;
+      },
+    })
+    .required(),
+});
+
+const ranksSchema = yup.object({
+  ID: yup.lazy(() => idSchema),
+  Title: yup.lazy(() => titleSchema),
+  MaxN: yup
+    .number()
+    .test({
+      name: 'compare-max-min',
+      message: 'Max/Min comparison failed',
+      test() {
+        const { path, parent } = this;
+        const { MinN, MaxN, ID } = parent;
+
+        if (!Number.isInteger(MaxN)) {
+          return this.createError({
+            path,
+            message: `Max should be an integer in ranks, in object ID: ${ID}`,
+          });
+        }
+
+        if (MaxN < 2) {
+          return this.createError({
+            path,
+            message: `Max should be higher or equal to 2 in ranks, in object ID: ${ID}`,
+          });
+        }
+
+        if (MaxN !== MinN) {
```

Comment:
I'm wondering then: does it make sense to have a `MaxN` and `MinN` on a `rank` question ?